src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -583,6 +584,12 @@ - (void)initStack
     factoryParams.otaProviderDelegate = sOTAProviderDelegate;
     factoryParams.shouldStartServer = YES;
 
+#ifdef DEBUG
+    // Force our controllers to only advertise on localhost, to avoid DNS-SD
+    // crosstalk.
+    [MTRDeviceController forceLocalhostAdvertisingOnly];
```

Comment:
Can we just move this into MTRTestCase setup (the `+` version, so it happens when any test suite inheriting from MTRTestCase starts)?

In particular, I think we should be doing this in any test that uses the "start a controller via the alloc/init API" setup, which is more than the two tests this PR touches.